src/main.cpp

```diff
@@ -2676,6 +2676,9 @@ bool static DisconnectTip(CValidationState& state, const Consensus::Params& cons
     // Write the chain state to disk, if necessary.
     if (!FlushStateToDisk(state, FLUSH_STATE_IF_NEEDED))
         return false;
+    if (fBare)
+        return true;
```

Comment:
I don't think we can return true here without calling `UpdateTip`?  Otherwise chainActive is not updated.